bedrock/firefox/templatetags/helpers.py

```diff
@@ -221,6 +221,66 @@ def download_firefox(ctx, channel='release', platform='all',
 
 
 @library.global_function
+@jinja2.contextfunction
+def download_firefox_desktop_list(ctx, channel='release', dom_id=None, locale=None,
```

Comment:
Looks like perhaps we can factor most of this into a function that both this and the `download_firefox` helper can use.